swarm/src/handler.rs

```diff
@@ -122,41 +123,91 @@ pub trait ConnectionHandler: Send + 'static {
     /// of simultaneously open negotiated inbound substreams. In other words it is up to the
     /// [`ConnectionHandler`] implementation to stop a malicious remote node to open and keep alive
     /// an excessive amount of inbound substreams.
+    #[deprecated(
+        since = "0.41.0",
+        note = "Handle `StreamEvent::FullyNegotiatedInbound` on `ConnectionHandler::on_event` instead.
+        the default of implemention of this inject_*` method delegates to it."
```

Comment:
```suggestion
        The default implementation of this `inject_*` method delegates to it."
```

Reply:
> Cool to see this work happening. Let us know once this is ready for a review.

Thanks Max, yeah I am waiting for #3011 to be merged cause this PR also touches some of the same files to then rebase against master and mark it as ready.